src/rocketsonde/core/probe.py

```diff
@@ -55,30 +67,26 @@ def probe_pid(self):
     def start_monitor(self):
         pid_queue = Queue(100)
         record_queue = Queue(10000)
-        shutdown_event = Event()
-        records_ready = Event()
-        process = Process(
-            target=self._worker_function, args=(self._metric, pid_queue, record_queue, records_ready, shutdown_event)
-        )
+        process = Process(target=self._worker_function, args=(self._metric, pid_queue, record_queue))
 
         process.start()
         atexit.register(self._panic_kill_monitor)
 
-        self._worker = _Worker(
-            shutdown_event=shutdown_event,
-            pid_queue=pid_queue,
-            record_queue=record_queue,
-            records_ready=records_ready,
-            process=process,
-        )
+        self._worker = _Worker(pid_queue=pid_queue, record_queue=record_queue, process=process)
 
     def _drain_record_pipe(self):
         if self._worker is not None:
-            while self._worker.records_ready.is_set() or not self._worker.record_queue.empty():
+            pipe_clean = False
+            while (self._worker.process.is_alive() and not pipe_clean) or not self._worker.record_queue.empty():
                 while not self._worker.record_queue.empty():
```

Comment:
Is there no timeout on this? That would add complexity though. Hmm.

Reply:
It's non-blocking

Reply:
That's good. Much less worry in that case.